docs/user_guide.md

```diff
@@ -1,17 +1,109 @@
+various underlying tasks, and which can therefore be considered as
+"black box" from a user perspective.
+
+Whenever new versions of the official WMO master BUFR tables are
+released by WMO (as noted above), they are downloaded and reformatted
+as new [NCEPLIBS-bufr master BUFR tables](@ref dfbfmstab) for use with
+the software and distributed with the next release of the
+library. However, users can also generate their own custom
+[NCEPLIBS-bufr DX BUFR tables](@ref dfbftab) for use with the software
+as needed.
+
+More details are available throughout the documentation, including
+sample utilities which demonstrate how to use many of the various
+library subroutines and functions to accomplish different tasks.
+
+@anchor hierarchy
+It's important to understand the following hierarchy:
+
+1. Any BUFR file can consist of one or more BUFR messages.
+2. Any BUFR message can consist of one or more BUFR data subsets, which are akin to reports from individual
+observational stations at a particular time and location.
+3. Any BUFR data subset can consist of one or more BUFR data descriptors and corresponding data values.
+
+## NCEPLIBS-bufr Library Subroutines and Functions
+
+This allows many of the most commonly used library subroutines and
+functions to be grouped, based on which level of the hierarchy they
+operate at, and whether for reading/decoding BUFR data or
+writing/encoding BUFR data:
+
+<table border>
+<tr>
+  <th>For working with:</th>
+  <th>BUFR files</th>
+  <th>BUFR messages</th>
+  <th>BUFR data subsets</th>
+  <th>BUFR data values</th>
+</tr>
+<tr>
+  <th>reading/decoding</th>
+  <td>openbf() openbf_f() ufbmem() ufbmex() cobfl()</td>
+  <td>readmg() ireadmg() ireadmg_f() readerme() readns() ireadns() rdmemm() readmm() ireadmm() crbmg() datelen() getlens() iupvs01() iupbs01() iupbs3() nmsub() rtrcpt() igetdate()</td>
+  <td>readsb() ireadsb() ireadsb_f() readns() ireadns() rdmems() ufbmms() ufbmns() ufbpos()</td>
+  <td>ufbint() ufbrep() ufbseq() ufbstp() ufbint_f() ufbrep_f() ufbrms() readlc() getvalnb() setbmiss() getbmiss() ibfms() icbfms()</td>
+</tr>
+<tr>
+  <th>writing/encoding</th>
+  <td>openbf() openbf_f() closbf() closbf_f() copybf() cobfl()</td>
+  <td>openmb() openmg() closmg() copymg() cpymem() cwbmg() cmpmsg() stdmsg() cnved4() strcpt() setblock() maxout() pkvs01() pkbs1() minimg()</td>
+  <td>writsb() writsa() writcp() copysb() icopysb()</td>
+  <td>ufbint() ufbrep() ufbseq() ufbstp() ufbint_f() ufbrep_f() writlc() setvalnb() setbmiss()</td>
+</tr>
+</table>
 
-The Python API documentation is available
-[here](https://noaa-emc.github.io/NCEPLIBS-bufr/python/index.html); however,
-note that this functionality is only available if the library was built using
-the -DENABLE_PYTHON=ON option during the cmake step.
+Many of the above subroutines and functions can also be accessed via a separate
+[Python API](https://noaa-emc.github.io/NCEPLIBS-bufr/python/index.html).
```

Comment:
This seems redundant - do we still need this here since you've already got a link to the same Python API at the top of this file?

In any case, I think the note about "this functionality is only available if the library was built using the -DENABLE_PYTHON=ON option during the cmake step." is useful and should probably be retained.

Reply:
Instead of saying that in the C/Fortran documentation, let's say that in the python documentation.

Honestly, no one will be surprised that the python package is optionally built. That's quite common.

I have removed the reference to the python docs at the bottom of the page.